app.js

```diff
@@ -1,18 +1,94 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
-
-// require spotify-web-api-node package here:
+const express = require("express");
+const hbs = require("hbs");
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
+
+app.use(express.urlencoded({ extended: true }));
 
 // setting the spotify-api goes here:
 
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
+
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (req, res) => {
+  res.render("search");
+});
+
+app.get("/artist-search", async (req, res) => {
+  const Sresponse = await spotifyApi
+    .searchArtists(req.query.name)
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+  const atistsItems = Sresponse.body.artists.items;
+  // console.log("The received data from the API: ", Sresponse.body);
+
+  // for (let item of atistsItems) {
+  //   console.log(item);
+  // }
+
+  const data = { doctitle: "Artists", artists: atistsItems };
+
+  res.render("artist-search-results", data);
+});
+
+app.get("/albums/:artistId", async (req, res) => {
+
+  const artistResponse = await spotifyApi.getArtist(req.params.artistId)
+  .catch((err) =>
+    console.log("The error while searching artists occurred: ", err)
+  );
+
+  const Sresponse = await spotifyApi
```

Comment:
the name Sresponse is weird. First off it starts with a capital letter - suggesting a class - second -> it's a kind of acronym, but not really... just destructure -> and get out items renamed as albums from the body ^^ and you don't have to worry about names haha